Detectors/CPV/reconstruction/src/RawDecoder.cxx

```diff
@@ -63,14 +63,16 @@ RawErrorType_t RawDecoder::readChannels()
       }
       skipUntilNextHeader = false;
       currentBC = header.bc();
+      currentOrbit = header.orbit();
       wordCountFromLastHeader = 0;
       nDigitsAddedFromLastHeader = 0;
-      if (currentOrbit != header.orbit()) { //bad cpvheader
-        LOG(ERROR) << "RawDecoder::readChannels() : "
-                   << "currentOrbit != header.orbit()";
-        mErrors.emplace_back(5, 0, 0, 0, kCPVHEADER_INVALID); //5 is non-existing link with general errors
-        skipUntilNextHeader = true;
-      }
+      // // There is no requirement that event orbit be equal to HBF orbit.
+      // if (currentOrbit != header.orbit()) { //bad cpvheader
```

Comment:
Hi @peressounko 
Why is it so? Isn't the payload after the RDH for given orbit supposed to correspond to triggers in this orbit?
I see that the RawFileWriter::addData is always called with the IR of the trigger, e.g.
https://github.com/AliceO2Group/AliceO2/blob/1bd90382249e1187060f52cc761c997025a9ceff/Detectors/CPV/simulation/src/RawWriter.cxx#L160
Is not your payload supposed to contain the same `trg.getBCData()`?

Reply:
Hi @shahor02 ,
What I see in RawFileWriter, first RDH header with some orbit is created.  Then payload is added with addData(...,IR,...) method where I provide IR. This IR is not used for header modification, but just for comparison that it is not too different from the initial RDH orbit. So I do not see where IR parameters corresponding to next payload are added to header.
Note that several events are normally written to one page, and here not only payload of the first event is checked, but all payloads in this page. And second and third should differ.